### PR TITLE
feat: add script to generate skip file

### PR DESCRIPTION
**Pull request by bajpai244**
# Pull Request type

Time spent on this PR:

Resolves: #626 

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Testing

## What is the new behavior?

This PR adds a new script to generate the `blockchain-tests-skip.yml` file from a `data.txt` { contains the logs of running the ef-tests } file in the project root.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -31,7 +31,7 @@ $(EF_TESTS_DIR):
 setup: $(EF_TESTS_DIR)
 
 setup-kakarot-v0: clean-kakarot-v0
-	@curl -sL -o kakarot-build.zip -H "Authorization: token $(GITHUB_TOKEN)" "$(KKRT_V0_BUILD_ARTIFACT_URL)"
+	@curl -sL -o kakarot-build.zip -H "Authorization: token $(GITHUB_TOKEN)" "$(KKRT_V0_BUILD_ARTIFACT_URL)'"
```
Comment (greged93) [inline Makefile:34]: typo?

File: scripts/generate_skip_file.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+with open("data.txt", "r") as f:
+    data = f.read()
+
+matches_failed = re.findall(r"thread '(.*)' panicked at ", data)
+matches_failed = matches_failed[1:]
+
+print(len(matches_failed))
+
+print(matches_failed[:10])
```
Comment (greged93) [inline scripts/generate_skip_file.py:12]: Can we remove this?

File: scripts/generate_skip_file.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+matches_failed = [(m.split('::')[-2].replace("_minus_", "-"), m.split('::')[-1]) for m in matches_failed]
```
Comment (greged93) [inline scripts/generate_skip_file.py]: I added 
```
    /// Formats the given string into a valid rust identifier.
    fn format_into_identifier(s: &str) -> String {
        s.replace('-', "_minus_")
            .replace('+', "_plus_")
            .replace('^', "_xor_")
    }
```
could you update this line to match

File: scripts/generate_skip_file.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
+#%%
```
Comment (greged93) [inline scripts/generate_skip_file.py]: This can go too

File: Makefile (hunk 1 above)
Comment (bajpai244) [reply Makefile]: fixed.

File: scripts/generate_skip_file.py (hunk 2 above)
Comment (bajpai244) [reply scripts/generate_skip_file.py]: fixed.

File: scripts/generate_skip_file.py (hunk 3 above)
Comment (bajpai244) [reply scripts/generate_skip_file.py]: fixed.

File: scripts/generate_skip_file.py (hunk 4 above)
Comment (bajpai244) [reply scripts/generate_skip_file.py]: fixed.

File: scripts/generate_skip_file.py [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+matches_failed = re.findall(r"thread '(.*)' panicked at ", data)
+matches_failed = matches_failed[1:]
+
+matches_failed = [
+    (
+        m.split("::")[-2]
+        .replace("-", "_minus_")
+        .replace("+", "_plus_")
+        .replace("^", "_xor_"),
```
Comment (greged93) [inline scripts/generate_skip_file.py:14]: should be the other way around for all of them

Comment (bajpai244) [reply scripts/generate_skip_file.py:14]: fixed.

**Comment by greged93** [comment]
you pushed data.txt and an update to the blockchain-skip.yml

**Review by greged93 — APPROVED** [review]
lgtm

**Pull request merged**